packages/cli/src/lib/setup/setupSetup.ts

```diff
@@ -130,41 +171,50 @@ function Setup(options) {
             console.error();
             console.error(COLOR_RESET);
         }
-        callback();
     }
 
     /**
      * Called after io-package objects are created
      *
-     * @param {Record<string, any>} systemConfig
-     * @param {() => void} callback
-     * @return {Promise<void>}
+     * @param systemConfig
+     * @param callback
      */
-    async function setupReady(systemConfig, callback) {
+    async setupReady(systemConfig: ConfigObject | undefined | null, callback: () => void): Promise<void> {
         if (!callback) {
-            console.log('database setup done. You can add adapters and start ' + tools.appName + ' now');
-            return processExit(EXIT_CODES.NO_ERROR);
+            console.log(`database setup done. You can add adapters and start ${tools.appName} now`);
+            return this.processExit(EXIT_CODES.NO_ERROR);
+        }
+
+        if (!this.objects) {
+            throw new Error('Objects not set up, call setupObjects first');
         }
 
         // clean up invalid user group assignments (non-existing user in a group)
         try {
-            await _cleanupInvalidGroupAssignments();
+            await this._cleanupInvalidGroupAssignments();
         } catch (e) {
             console.error(`Cannot clean up invalid user group assignments: ${e.message}`);
         }
 
-        if (!objects.syncFileDirectory || !objects.dirExists) {
-            return void informAboutPlugins(systemConfig, callback);
+        // special methods which are only there on objects server
+        // TODO this check will lead to objects being never in the following code
+        if (!('syncFileDirectory' in this.objects) || !('dirExists' in this.objects)) {
```

Comment:
Maybe doing `this.objects as any` here prevents the problem?

Reply:
I guess in the long run we could add the methods as optional to the db interface, then it could work like that?

Reply:
Yeah probably